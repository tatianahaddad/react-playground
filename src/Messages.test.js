import React from 'react';
import ReactDOM from 'react-dom';
import Messages from './Messages';
import renderer from 'react-test-renderer';

describe ('Messages components', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Messages />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Messages name="Messages" unread={4}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  
    it('renders the UI as expected for 0', () => {
      const treeTwo = renderer
        .create(<Messages name="Messages" unread={0}/>)
        .toJSON();
      expect(treeTwo).toMatchSnapshot();  
  });
})


