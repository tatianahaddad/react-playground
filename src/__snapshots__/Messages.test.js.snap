// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages components renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count"
  >
    4
  </div>
</div>
`;

exports[`Messages components renders the UI as expected for 0 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;

exports[`renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count"
  >
    4
  </div>
</div>
`;

exports[`renders the UI as expected for 0 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;
